### PR TITLE
add listener manager to store listener

### PR DESCRIPTION
**Pull request by wackxu**
Signed-off-by: wackxu <xushiwei5@huawei.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

we want to sync watch request between cloud and edge, so we cache the listener by nodeID and gvr so that we can
compare the difference between cloud and edge easilly

**Which issue(s) this PR fixes**:

Fixes # part of https://github.com/kubeedge/kubeedge/issues/4423

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cloud/pkg/dynamiccontroller/application/listener_manager.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	listeners, exists = lm.listenerByGVR[listener.gvr]
+	if exists {
+		delete(listeners, listener.id)
+		if len(lm.listenerByGVR[listener.gvr]) == 0 {
+			delete(lm.listenerByGVR, listener.gvr)
+		}
+	}
+}
+
+func (lm *listenerManager) GetListenersForNode(nodeName string) map[string]*SelectorListener {
```
Comment (vincentgoat) [inline cloud/pkg/dynamiccontroller/application/listener_manager.go:83]: Seems the function `GetListenersForNode` is not called except for the test file.

Comment (wackxu) [reply cloud/pkg/dynamiccontroller/application/listener_manager.go:83]: this will be used in the follow PR

Comment (wackxu) [reply cloud/pkg/dynamiccontroller/application/listener_manager.go:83]: please see https://github.com/kubeedge/kubeedge/pull/4484

**Pull request merged**